test/integration/model.test.js

```diff
@@ -1698,10 +1698,81 @@ describe(Support.getTestDialectTeaser('Model'), () => {
+      });
+
+      it('returns the affected rows count if `options.returning` is set to false', async function () {
+        await this.User.bulkCreate(data);
+        let affectedRowsCount = await this.User.destroy({ where: { secretValue: '42' }, returning: false });
+        expect(affectedRowsCount).to.be.a('number');
+        expect(affectedRowsCount).to.equal(2);
+        affectedRowsCount = await this.User.destroy({ where: {}, returning: false });
+        expect(affectedRowsCount).to.be.a('number');
+        expect(affectedRowsCount).to.equal(3);
+      });
+
+      it('returns only the fields in affected rows specified in `options.returning`', async function () {
+        await this.User.bulkCreate(data);
+        let [affectedRowsCount, affectedRows] = await this.User.destroy({ where: { secretValue: '42' }, returning: ['username'] });
+        expect(affectedRowsCount).to.be.a('number');
+        expect(affectedRowsCount).to.equal(2);
+        expect(affectedRows).to.be.an('array').to.have.lengthOf(2);
+        expect(affectedRows[0]).to.be.instanceOf(this.User);
+        JSON.parse(JSON.stringify(affectedRows)).forEach(userObj => {
+          expect(userObj).to.have.all.keys('username');
+          expect(userObj).to.not.have.keys('id', 'secretValue');
+        });
+        [affectedRowsCount, affectedRows] = await this.User.destroy({ where: {}, returning: ['*'] });
+        expect(affectedRowsCount).to.be.a('number');
+        expect(affectedRowsCount).to.equal(3);
+        expect(affectedRows).to.be.an('array').to.have.lengthOf(3);
+        expect(affectedRows[0]).to.be.instanceOf(this.User);
+        JSON.parse(JSON.stringify(affectedRows)).forEach(userObj => {
+          expect(userObj).to.include.keys('secretValue', 'username', 'id');
+        });
+      });
+
+      it('returns all the fields in affected rows when `options.returning` is set to []', async function () {
+        await this.User.bulkCreate(data);
+        let [affectedRowsCount, affectedRows] = await this.User.destroy({ where: { secretValue: '42' }, returning: [] });
+        expect(affectedRowsCount).to.be.a('number');
+        expect(affectedRowsCount).to.equal(2);
+        expect(affectedRows).to.be.an('array').to.have.lengthOf(2);
+        expect(affectedRows[0]).to.be.instanceOf(this.User);
+        JSON.parse(JSON.stringify(affectedRows)).forEach(userObj => {
+          expect(userObj).to.include.keys('secretValue', 'username', 'id');
+        });
+        [affectedRowsCount, affectedRows] = await this.User.destroy({ where: {}, returning: [] });
+        expect(affectedRowsCount).to.be.a('number');
+        expect(affectedRowsCount).to.equal(3);
+        expect(affectedRows).to.be.an('array').to.have.lengthOf(3);
+        expect(affectedRows[0]).to.be.instanceOf(this.User);
+        JSON.parse(JSON.stringify(affectedRows)).forEach(userObj => {
+          expect(userObj).to.include.keys('secretValue', 'username', 'id');
+        });
+      });
+    }
```

Comment:
Can you add a test that ensures an error is thrown if the dialect uses 'returning', but it isn't supported in that dialect?

```suggestion
    } else {
      it('throws an error when "returning" is used in an unsupported dialect', () => {
        await expect(this.User.destroy({ returning: true })).to.be.rejectedWith(`${dialect} does not support the "returning" option in DELETE queries.`);
      });
    }
```